### ISSUE TITLE
Is possible use Zimbra Uid instead Zimbra Id?

### ISSUE THREAD

**Issue by dcna-io** [ZeXtras/zimbra-drive#21]
Hi,

I have a Zimbra Community Edition 8.8 and an Owncloud server, both authenticate in Active Directory. When I try to use Zimbra Drive, the extension creates a new user using Zimbra Id and not bind with the exists user in Owncloud or AD, that use the same name of Zimbra Uid.

**Comment by ZxUgoPadoan** [comment]
Hi @dcna-io 
by using Zimbra Uid, you can't have more than one domain (two account of different domain can have the same Uid).
If you want, you can modify 
zimbra-extension/src/com/zextras/zimbradrive/NcUserZimbraBackendHttpHandler.java:199
and call 'account.getUId()' instead of 'account.getId()'.
To build the extension, go to zimbra-extension folder and run 'make all', the new extension should appear in zimbra-extension/dist.

**Comment by TomislavNovosel** [comment]
Hello,
we have the similar problem and are using zimbra 8.8.6.
I am unable to find zimbra-extensions folder? If i search trough zimbra folder, i have only zimlets-deployed folder? In there i can see com_zextras_drive_open, and there are several .js files in there. I have searched trough them, but no mention of 'account.getId()' function.
Any ideas?

**Comment by ZxUgoPadoan** [comment]
Hi @TomislavNovosel,
in the previous message I was referring to this repository (https://github.com/ZeXtras/zimbra-drive), no to a Zimbra instance.

**Comment by TomislavNovosel** [comment]
Hello @ZxUgoPadoan ,

ok, i understand now.
I have installed git, and cloned this repository locally. I found and changed file... but how do i run 'make all'? I am git beginner so please bee patient...

Thanks

**Comment by ZxUgoPadoan** [comment]
Hi @TomislavNovosel ,
make sure to have GNU Make installed in you system.
Open the command line and change the directory to 'zimbra-extension' folder (in your cloned repository), now run the command 'make  all'.

Have a nice day.

**Comment by TomislavNovosel** [comment]
Hello @ZxUgoPadoan ,

i installed make, but when i run it get following error:
tomislav.novosel@LAPTOP-B7HJRUS MINGW64 ~/Documents/GitHub/zimbra-drive/zimbra-extension (master)
$ make all
mkdir -p lib
wget http://central.maven.org/maven2/commons-io/commons-io/1.4/commons-io-1.4.jar \
        -O lib/commons-io-1.4.jar
process_begin: CreateProcess(NULL, wget http://central.maven.org/maven2/commons-io/commons-io/1.4/commons-io-1.4.jar -O lib/commons-io-1.4.jar, ...) failed.
make (e=2): The system cannot find the file specified.
make: *** [Makefile:25: lib/commons-io-1.4.jar] Error 2

**Comment by TomislavNovosel** [comment]
Got it.
Had to do it on a linux machine... works like a charm.
Thank you.

**Comment by franklupo** [comment]
Hi,
excellent work. It is not possible to create a property file where to specify what to use?

example

> config.properties
> accountId=Id
> ...
> 
> Id,UId,MailmName
> 
> String key = prop.getProperty("accountId");
> String accountId=account.getId();
> if(key == "Id"){
>   accountId=account.getId();
> }
> else if(key == "UId"){
>   accountId=account.getUId();
> }
> ....
> userAttributesJson.put("accountId", accountId);

Best regards

**Comment by FredNass** [comment]
Hi ZxUgoPadoan,

"If you want, you can modify zimbra-extension/src/com/zextras/zimbradrive/NcUserZimbraBackendHttpHandler.java:199 and call 'account.getUId()' instead of 'account.getId()'."

Does that work for any other Zimbra ldap attribute ?

I mean if I sync the Zimbra LDAP 'co' (friendly country name) attribute with my company LDAP 'uid' attribute, can I just change 'account.getId()' with 'account.getCo()' and the extension will pass the 'co' attribute to owncloud / nextcloud servers ?

I'm trying to figure out a way for the zimbradrive-extension to bind to the nextcloud servers with an attribute that I could set based on our company LDAP.

Regards, (fantastic work by the way :-1: )

Frédéric.

**Comment by FredNass** [comment]
Apparently, it uses ZAL but I can't see anythig related to the 'co' attribute in the ZAL source code. Would that be hard to code ?

**Comment by FredNass** [comment]
It would be so cool if we could set this out of the code like a zimbraDriveAccountIdAttribute attribute.

**Comment by ccasalicchio** [comment]
I agree with @FredNass , because currently, when clicking on the Drive tab from within Zimbra, it creates an awful username based of the UUID (something like 71b2c6cd-27fb-44c3-bf97-161c7a0dccc1) that makes it impractical to manage with the owncloud desktop sync or even to manage users accessing owncloud directly from the web interface (and not from zimbra webmail). I have currently turned off zimbra drive to avoid users creating accounts in Owncloud with those awful usernames. Besides, zimbra drive does not "see" any shared folder... o well, hopefully this changes in future versions

**Comment by Piero3** [comment]
Hello !
It would be a good thing if zimbradrive also pushed the mailing lists as a group in nextcloud. The best thing would be to be able to finely configure the request as with a (ldap backend) to filter the attributes that you want to use.
